test/controllers/passwordless/sessions_controller_test.rb

```diff
@@ -12,6 +12,12 @@ def create_session_for(user)
       )
     end
 
+    def User.fetch_resource_for_passwordless(email)
+      return if email == 'invalidemail'
+
+      User.find_or_create_by(email: email)
+    end
```

Comment:
Hmm, should we undefine it after this spec? This seems like something that could give some pretty weird bugs later in the test suite.

Reply:
Yeah. I've moved the definition inside the respective test, and cleaning it up after